r1cs/gadgets/crypto/src/commitment/pedersen/mod.rs

```
@@ -60,8 +63,14 @@ where
             }
         }
 
-        assert_eq!(padded_input.len() * 8, W::WINDOW_SIZE * W::NUM_WINDOWS);
-        assert_eq!(parameters.params.generators.len(), W::NUM_WINDOWS);
+        // TODO: check error message
+        if padded_input.len() * 8 != W::WINDOW_SIZE * W::NUM_WINDOWS {
+            return Err(SynthesisError::Other("padded input length verification failed".to_owned()));
```

Let's return the same error message we returned for the primitive